docs/static/troubleshooting.asciidoc

```diff
@@ -139,9 +137,38 @@ For general performance tuning tips and guidelines, see <<performance-tuning>>.
 
 
 
+[float] 
+[[ts-pipeline]] 
+== Troubleshooting a pipeline
+
+Pipelines, by definition, are unique. Here are some guidelines to help you get
+started.
+
+* Identify the offending pipeline.
+* Start small. Create a minimum pipeline that manifests the problem.
 
 
+For basic pipelines, this configuration could be enough to make the problem show itself.
 
+[source,ruby]
+-----
+input {stdin{}} output {stdout{}}
+-----
+
+To better identify the offending pipeline, {ls} could separate its logs by pipeline.
+To switch on the log per pipeline feature is sufficient to enable `pipeline.separate_logs` in your `logstash.yml`
+or passing the `-Dls.pipeline.separate_logs=true` on command line.
```

Comment:
I'm not sure this is quite right, or if we even _can_ change this setting from the command line.

Although we use the Java System Property `ls.pipeline.separate_logs` in the log4j templates (and one would ordinarily set Java System Properties with `-Dproperty=value`), it looks like on startup we stomp on the existing value of that property without first reading it, stetting to to be the value held in `pipeline.separate_logs` from `logstash.yml`:

> ~~~ ruby
>     java.lang.System.setProperty("ls.pipeline.separate_logs", setting("pipeline.separate_logs").to_s)
> ~~~
> -- [logstash-core/lib/logstash/runner.rb:257@2925b491](https://github.com/elastic/logstash/blob/2925b491f9a68afc0277e35bdd6fe07ea17b4eeb/logstash-core/lib/logstash/runner.rb#L257)

Reply:
@yaauie you are right, many thanks to point it out. Removed and rephrased a little bit